src/test_app.rs

```diff
@@ -154,3 +137,10 @@ impl ServerTestAppExt for App {
         })
     }
 }
+
+/// Stores connected client entity from server on client.
+///
+/// Inserted in [`ServerTestAppExt::connect_client`] and removed by [`ServerTestAppExt::disconnect_client`].
+/// Used to track which client corresponds to which connection.
+#[derive(Resource, Deref, Clone, Copy, Debug)]
+pub struct TestClientEntity(Entity);
```

Comment:
Since clients no longer store `ClientId` in connected state, I need some mechanism to identify which client app belongs to which connected entity. This is why I introduced this resource.